File: backend/src/main/java/com/example/Flicktionary/domain/actor/entity/Actor.java
package com.example.Flicktionary.domain.actor.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Actor {
    @Id
    private Long id;

    @Column(nullable = false)
    private String name;

    private String profilePath;
}

File: backend/src/main/java/com/example/Flicktionary/domain/actor/entity/Actor.kt
package com.example.Flicktionary.domain.actor.entity

import jakarta.persistence.*

@Entity
class Actor(
    @Column(nullable = false)
    val name: String,

    val profilePath: String?
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private var _id: Long? = null

    var id: Long
        get() = _id ?: 0
        set(value) {
            _id = value
        }
}